src/page/mod.rs

```diff
@@ -43,6 +43,9 @@ pub trait Materializer {
 /// Points to either a memory location or a disk location to page-in data from.
 #[derive(Debug, Clone, PartialEq)]
 pub enum CacheEntry<M: Send + Sync> {
+    /// A cache item that contains the most recent fully-merged page state, also in secondary
+    /// storage.
+    MergedResident(M, LogID),
```

Comment:
this allows us to do less cpu-intensive merging when retrieving pages, at the cost of higher contention for mutations

Reply:
in the future it may be possible to do this without effecting contention by making pages mergeable without effecting the stack pointers